modules/ding_campaign_plus/includes/ding_campaign_plus.admin.inc

```diff
@@ -0,0 +1,204 @@
+}
+
+/**
+ * Build options (table rows) for the administration UI.
+ *
+ * @param array $weights
+ *   The weights used to determine the sort order of the options.
+ *
+ * @return array
+ *   The options sorted by weight.
+ */
+function ding_campaign_plus_ding_settings_options(array $weights) {
+  $options = array();
+
+  foreach (module_implements('ding_campaign_plus_default_weights') as $module) {
+    $types = module_invoke($module, 'ding_campaign_plus_default_weights');
+    foreach ($types as $type => $info) {
+      $options[$type] = array(
+        'title' => $info['prefix'] . ' - ' . $info['title'],
+        'weight' => isset($weights[$type]) ? $weights[$type] : $info['weight'],
+      );
+    }
+  }
+
+  $options += array(
+    'search' => array(
+      'title' => 'Search (CQL)',
+      'weight' => isset($weights['object']) ? $weights['object'] : 0,
+    ),
+  );
+
+  // Sort options be weight keeping the array keys.
+  uasort($options, '_ding_campaign_plus_ding_settings_options_sort');
+
+  return $options;
+}
+
+/**
+ * Compare function use to sort options.
+ *
+ * @param array $a
+ *   Single option.
+ * @param array $b
+ *   Single option.
+ *
+ * @return int
+ *   The result of the comparison.
+ */
+function _ding_campaign_plus_ding_settings_options_sort(array $a, array $b) {
+  if ($a['weight'] === $b['weight']) {
+    return 0;
+  }
+  return ($a['weight'] < $b['weight']) ? -1 : 1;
+}
+
+/**
+ * Build single row for use in the table.
+ *
+ * @param string $title
+ *   The title to display in the table.
+ * @param int $weight
+ *   The weight of the row in the table.
+ *
+ * @return array
+ *   Form element for the table.
+ */
+function ding_campaign_plus_admin_settings_form_add($title, $weight = 0) {
+  return array(
+    // We'll use a form element of type '#markup' to display the item name.
+    'name' => array(
+      '#markup' => check_plain($title),
+    ),
+
+    'weight' => array(
+      '#type' => 'weight',
+      '#title' => t('Weight'),
+      '#default_value' => $weight,
+      '#delta' => 10,
+      '#title_display' => 'invisible',
+    ),
+  );
+}
+
+/**
+ * Implements theme_ding_campaign_plus_admin_form().
+ *
+ * Themes the admin forms weight table.
+ */
+function theme_ding_campaign_plus_admin_form($variables) {
+  $form = $variables['form'];
+
+  $rows = array();
+  foreach (element_children($form['ding_campaign_plus_weight']['table']) as $id) {
+    $form['ding_campaign_plus_weight']['table'][$id]['weight']['#attributes']['class'] = array('campaign-type-weight');
+    $rows[] = array(
+      'data' => array(
+        drupal_render($form['ding_campaign_plus_weight']['table'][$id]['name']),
+        drupal_render($form['ding_campaign_plus_weight']['table'][$id]['weight']),
+      ),
+      'class' => array('draggable'),
+    );
+  }
+
+  $header = array(t('Name'), t('Weight'));
+  $table_id = 'campaign-types-table';
+
+  $output = theme('table', array(
```

Comment:
Calling the theme() function directly is strongly discouraged. See:
https://api.drupal.org/api/drupal/includes%21theme.inc/function/theme/7.x
Also, there's no need to theme the table before outputing it - D7 handles render arrays very well.

Reply:
Yes, but as I understanden the core documentation and the example module. This is the only way to make a table with table-drag.

See https://cgit.drupalcode.org/examples/tree/tabledrag_example/tabledrag_example_simple_form.inc?h=7.x-1.x

Reply:
That's how it's mostly done in examples, yes. But apparently it's not necessary to theme the table first. 
See f.ex.: https://gist.github.com/AlexMazaltov/096a525044c85c2f255a

Reply:
hmm, will look into that

Reply:
I have tried the way now and can't get it to work. The "cross" will simply not show up. Write to you about this on slack.